app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationAnimationActivity.kt

```diff
@@ -0,0 +1,1783 @@
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.testapp.databinding.ActivityViewAnnotationAnimationBinding
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.turf.TurfConstants.UNIT_DEFAULT
+import com.mapbox.turf.TurfMeasurement
+import kotlin.math.roundToLong
+
+/**
+ * Example how to animate a view annotations on top of a route.
+ */
+class ViewAnnotationAnimationActivity : AppCompatActivity() {
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private var routeCoordinateList = mutableListOf<Point>()
+  private lateinit var button: Button
+  private var totalLength = 0.0
+  private var count = 0
+  private var currentAnimator: Animator? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewAnnotationAnimationBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    // calculate the route length, get coordinates from route geometry
+    val featureCollection = FeatureCollection.fromJson(ROUTE_GEOJSON)
+    val lineString = featureCollection.features()!![0].geometry() as LineString
+    routeCoordinateList = lineString.coordinates()
+    totalLength = TurfMeasurement.length(lineString, UNIT_DEFAULT)
+
+    // initialize the mapview
+    binding.mapView.getMapboxMap().apply {
+      loadStyle(
+        style(Style.LIGHT) {
+          // source for displaying the route
+          +geoJsonSource(SOURCE_ID) {
+            featureCollection(featureCollection)
+          }
+          // layer for displaying the route
+          +lineLayer(LINE_ID, SOURCE_ID) {
+            lineColor(Color.parseColor("#F13C6E"))
+            lineWidth(4.0)
+          }
+        }
+      ) {
+        // center camera around SF airport
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(-122.3915, 37.6177))
+            .zoom(11.0)
+            .build()
+        )
+
+        // create a dummy view to animate
+        button = Button(binding.mapView.context)
+        button.layoutParams = FrameLayout.LayoutParams(WRAP_CONTENT, WRAP_CONTENT)
+
+        // add the dummy view to the map as view annotation
+        viewAnnotationManager = binding.mapView.viewAnnotationManager
+        viewAnnotationManager.addViewAnnotation(
+          button,
+          ViewAnnotationOptions.Builder()
+            .geometry(routeCoordinateList[0])
+            .build()
+        )
+        animateNextStep()
+      }
+    }
+  }
+
+  override fun onStop() {
+    super.onStop()
+    currentAnimator?.cancel()
+  }
+
+  private fun animateNextStep() {
+    if (count == routeCoordinateList.size - 1) {
+      count = 0
+    }
+    val currentPoint = routeCoordinateList[count]
+    val nextPoint = routeCoordinateList[count + 1]
+    val animateDuration = TurfMeasurement.distance(currentPoint, nextPoint) / totalLength * 30000
```

Comment:
30000 to const and give meaningful name?

Reply:
:+1: first part should be renamed to progress and 30000 should be renamed to total animation length